test/testdata/rbi/dir.rb

```diff
@@ -0,0 +1,9 @@
+# typed: strict
+
+extend T::Sig
+
+sig {returns(T::Array[String])}
+def test_dir_brackets
+  Dir['a', 'b']
+  Dir['a', base: '.']
```

Comment:
Might be best to also test for `Pathname` values and supplying a block.

Reply:
I added a test with a block without adding a `blk` parameter to the rbi and the test and type checking passed 🧐 what can I conclude from this? Is the alternative signature with a block parameter not necessary?

Reply:
In Ruby, any method could be supplied a block, even if it does not declare one. You can call those implicit blocks using `yield`. Thus, when you supply a block to `Dir.[]`, Sorbet thinks that is the case and does not complain. However, if you inspect the block argument type, you will find it to be `T.untyped`.

The block for `Dir.[]` is optional and sends a `String` as the block argument, so declaring a `T.nilable(T.proc.params(arg0: String).void)` (nilable is to signify that it is optional) block parameter would ensure the correct signature for the block that is passed. You can assert the type of the block argument in the test using `T.assert_type!` to ensure that it is a `String`.